### PR TITLE
BUG: linalg: for BLAS, downcast complex256 to complex128, not float64

### LINKED ISSUE #6194: lstsq() and others detect numpy.complex256 as real
For example:
```
>>> A = np.array([[1,2,3],[4,5,6],[7,8,9]])
>>> b = [3,2,1]
>>> dt=np.complex256; la.lstsq(array(A, dtype=dt), array(b, dtype=dt), lapack_driver='gelsd')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-45-1282c50dbc51> in <module>()
----> 1 dt=np.complex256; la.lstsq(array(A, dtype=dt), array(b, dtype=dt), lapack_driver='gelsd')

/usr/lib/python3.5/site-packages/scipy/linalg/basic.py in lstsq(a, b, cond, overwrite_a, overwrite_b, check_finite, lapack_driver)
    867         elif driver == 'gelsd':
    868             if real_data:
--> 869                 lwork, iwork = _compute_lwork(lapack_lwork, m, n, nrhs, cond)
    870                 if iwork == 0:
    871                     # this is LAPACK bug 0038: dgelsd does not provide the

ValueError: too many values to unpack (expected 2)
```

Choosing a driver that skips this check works fine, casting the complex256 to a complex128 along the way as expected:

```
>>> dt=np.complex256; la.lstsq(array(A, dtype=dt), array(b, dtype=dt), lapack_driver='gelsy')

(array([-1.94444444+0.j, -0.11111111+0.j,  1.72222222+0.j]),
 array([], dtype=complex128),
 2,
 None)
```

Tracing through this got me to line 217 of scipy/linalg/blas.py, which should probably change from
`    if dtype.char not in 'fdFD':`
to
`    if dtype.char not in 'fdgFDG':`
unless that's going to break something else.

### PR DESCRIPTION
**Pull request by ev-br**
closes https://github.com/scipy/scipy/issues/6194

### CODE REVIEW

File: scipy/linalg/blas.py [hunk 1]
```diff
@@ -214,7 +214,10 @@ def find_best_blas_type(arrays=(), dtype=None):
             prefer_fortran = True
 
     prefix = _type_conv.get(dtype.char, 'd')
-    if dtype.char not in 'fdFD':
```
Comment (pv) [inline scipy/linalg/blas.py]: The conversion needs to be consistent with the `prefix` character.

File: scipy/linalg/tests/test_basic.py [hunk 2]
```diff
@@ -40,11 +40,21 @@
...
+COMPLEX_DTYPES = [np.complex64, np.complex128, np.complex256]
 DTYPES = REAL_DTYPES + COMPLEX_DTYPES
 
 
+def _eps_cast(dtyp):
+    """Get the epsilon for dtype, possibly downcast to BLAS types."""
+    dt = dtyp
+    if dt == np.float128:
+        dt = np.float64
+    elif dt == np.complex256:
```
Comment (pv) [inline scipy/linalg/tests/test_basic.py]: complex256 and float128 do not exist on all platforms (on some the sizes are complex192 and float96 etc.). The "invariant" names are iirc np.longdouble, np.clongdouble

File: scipy/linalg/blas.py (hunk 1 above)
Comment (ev-br) [inline scipy/linalg/blas.py]: This seems to be the case already: https://github.com/scipy/scipy/pull/6207/files#diff-51d8d45efc9d4bd7634e0d31ee815602L167

File: scipy/linalg/tests/test_basic.py (hunk 2 above)
Comment (ev-br) [inline scipy/linalg/tests/test_basic.py]: yuk. Updated.

**Pull request merged**